superset/utils/pandas_postprocessing.py

```diff
@@ -544,3 +563,105 @@ def contribution(
     if temporal_series is not None:
         contribution_df.insert(0, DTTM_ALIAS, temporal_series)
     return contribution_df
+
+
+def prophet(  # pylint: disable=too-many-arguments,too-many-locals
```

Comment:
Could this function be broken up? Generally comments in a large function like this are a good place to draw divisions

Reply:
These unfortunately need to be one single function with `too-many-arguments`, as they are called based on instructions provided in the chart data request. But you're right, `too-many-locals` doesn't need to be part of the design here, so will try to break out some internal logic.